[DOC] Update changelog (#3020)
seqan/seqan3: CHANGELOG.md

## New features

#### Alphabet

>   * `cigar` can now be assigned from `std::string_views` ([\#2966](https://github.com/seqan/seqan3/pull/2966)).

Reviewer: ```suggestion
  * `seqan3::cigar` can now be assigned from `std::string_views` ([\#2966](https://github.com/seqan/seqan3/pull/2966)).
```

  * Added `seqan3::views::char_strictly_to`. Behaves like `seqan3::views::char_to`, but throws on invalid
    input ([\#2898](https://github.com/seqan/seqan3/pull/2898)).

#### I/O
 * Added an option to ignore blanks before IDs when reading FASTA-files
   ([\#2770](https://github.com/seqan/seqan3/pull/2770)).
 * Added functionality to store a file offset inside alignment files
   ([\#2501](https://github.com/seqan/seqan3/pull/2501)).

#### Utility
  * Added `seqan3::list_traits::repeat` ([\#2899](https://github.com/seqan/seqan3/pull/2899)).

## Notable Bug-fixes

#### Core
  * Added missing implementations for AVX512 ([\#2920](https://github.com/seqan/seqan3/pull/2920) and
    [\#2926](https://github.com/seqan/seqan3/pull/2926)).

#### IO

>   * Resolved an issue parsing FASTA files containing sequence IDs that start with `>`

Reviewer: which issue? This is really abstract

    ([\#2869](https://github.com/seqan/seqan3/pull/2869)).

#### Search
  * Relaxed `kmer_hash_view::iterator` difference requirement ([\#2931](https://github.com/seqan/seqan3/pull/2931)).

>   * Improved performance of `seqan3::counting_vector::operator+=` ([\#2930](https://github.com/seqan/seqan3/pull/2930)).

Reviewer: isn't this rather a feature?

## New features

#### Alphabet
  * `seqan3::cigar` can now be assigned from `std::string_view` ([\#2966](https://github.com/seqan/seqan3/pull/2966)).
  * Added `seqan3::views::char_strictly_to`. Behaves like `seqan3::views::char_to`, but throws on invalid
    input ([\#2898](https://github.com/seqan/seqan3/pull/2898)).

#### I/O

>  * Added an option to ignore blanks before IDs when reading FASTA-files

Reviewer: ```suggestion
 * Added `seqan3::sequence_file_option::fasta_ignore_blanks_before_id` to ignore blanks before IDs when reading FASTA-files (e.g. `>       some_id` will store only `"some_id"` as id.)
```

  * Replaced `seqan3::views::to` (implemented via range-v3) with `seqan3::ranges::to` (implemented in SeqAn3).
    `seqan3::ranges::to` provides a subset of C++23's `std::ranges::to` and will be replaced with the STL-equivalent

>     in a future version ([\#2969](https://github.com/seqan/seqan3/pull/2969)).

Reviewer: ```suggestion
    in a future version ([\#2969](https://github.com/seqan/seqan3/pull/2969)).
  * Replaced the implementation of `seqan3::views::chunk`. It is now implemented in SeqAn3 and does not use 
    the range-v3 implementation anymore. ([\#2975](https://github.com/seqan/seqan3/pull/2975))
```
